src/plugins/discover/public/application/components/context_app/use_context_app_fetch.tsx

```diff
@@ -0,0 +1,185 @@
+import { AppState } from '../../angular/context_state';
+import { getFirstSortableField } from '../../angular/context/api/utils/sorting';
+
+const createError = (statusKey: string, reason: FailureReason, error?: Error) => ({
+  [statusKey]: { value: LoadingStatus.FAILED, error, reason },
+});
+
+export interface ContextAppFetchProps {
+  anchorId: string;
+  indexPatternId: string;
+  indexPattern: IndexPattern;
+  appState: AppState;
+  useNewFieldsApi: boolean;
+  services: DiscoverServices;
+}
+
+export function useContextAppFetch({
+  anchorId,
+  indexPatternId,
+  indexPattern,
+  appState,
+  useNewFieldsApi,
+  services,
+}: ContextAppFetchProps) {
+  const { uiSettings: config, data, indexPatterns, toastNotifications, filterManager } = services;
+
+  const searchSource = useMemo(() => {
+    return data.search.searchSource.createEmpty();
+  }, [data.search.searchSource]);
+  const tieBreakerField = useMemo(
+    () => getFirstSortableField(indexPattern, config.get(CONTEXT_TIE_BREAKER_FIELDS_SETTING)),
+    [config, indexPattern]
+  );
+  const fetchAnchor = useMemo(() => {
+    return fetchAnchorProvider(indexPatterns, searchSource, useNewFieldsApi);
+  }, [indexPatterns, searchSource, useNewFieldsApi]);
+  const { fetchSurroundingDocs } = useMemo(
+    () => fetchContextProvider(indexPatterns, useNewFieldsApi),
+    [indexPatterns, useNewFieldsApi]
+  );
+
+  const [fetchedState, setFetchedState] = useState<ContextFetchState>(
+    getInitialContextQueryState()
+  );
+
+  const setState = useCallback((values: Partial<ContextFetchState>) => {
+    setFetchedState((prevState) => ({ ...prevState, ...values }));
+  }, []);
+
+  const fetchAnchorRow = useCallback(async () => {
+    const { sort } = appState;
+    const [[, sortDir]] = sort;
+    const errorTitle = i18n.translate('discover.context.unableToLoadAnchorDocumentDescription', {
+      defaultMessage: 'Unable to load the anchor document',
+    });
+
+    if (!tieBreakerField) {
+      setState(createError('anchorStatus', FailureReason.INVALID_TIEBREAKER));
+      toastNotifications.addDanger({
+        title: errorTitle,
+        text: toMountPoint(
+          <MarkdownSimple>
+            {i18n.translate('discover.context.invalidTieBreakerFiledSetting', {
+              defaultMessage: 'Invalid tie breaker field setting',
+            })}
+          </MarkdownSimple>
+        ),
+      });
+      return;
+    }
+
+    try {
+      setState({ anchorStatus: { value: LoadingStatus.LOADING } });
+      const anchor = await fetchAnchor(indexPatternId, anchorId, [
+        fromPairs(sort),
+        { [tieBreakerField]: sortDir },
+      ]);
+      setState({ anchor, anchorStatus: { value: LoadingStatus.LOADED } });
+      return anchor;
+    } catch (error) {
+      setState(createError('anchorStatus', FailureReason.UNKNOWN, error));
+      toastNotifications.addDanger({
+        title: errorTitle,
+        text: toMountPoint(<MarkdownSimple>{error.message}</MarkdownSimple>),
```

Comment:
Since this is displayed to the user, let's log the error message and put something more user-friendly in the UI.

Reply:
Assume we can't really provide more user friendly message to ui, since it's unknown error which might be caused a bad network, bad es response... one meaningful message we already providing is localised `errortitle`